tests/NLog.UnitTests/ConfigFileLocatorTests.cs

```diff
@@ -48,9 +45,234 @@ namespace NLog.UnitTests
+            {
+                AppDomainBaseDirectory = Path.Combine(tmpDir, "BaseDir"),
+#if NETSTANDARD
+                AppDomainConfigurationFile = string.Empty,                  // NetCore style
+#else
+                AppDomainConfigurationFile = Path.Combine(tmpDir, "EntryDir", "Entry.exe.config"),
+#endif
+                CurrentProcessFilePath = Path.Combine(tmpDir, "ProcessDir", "Process.exe"),// NetCore dotnet.exe
+                EntryAssemblyLocation = Path.Combine(tmpDir, "EntryDir"),
+                EntryAssemblyFileName = "Entry.dll"
+            };
+
+            var fileLoader = new LoggingConfigurationFileLoader(appEnvMock);
+
+            // Act
+            var result = fileLoader.GetDefaultCandidateConfigFilePaths().ToList();
+
+            // Assert base-directory + entry-directory + nlog-assembly-directory
+            if (NLog.Internal.PlatformDetector.IsWin32)
+            {
+#if NETSTANDARD
+                Assert.Equal(5, result.Count);  // Case insensitive
+#else
+                Assert.Equal(4, result.Count);  // Case insensitive
+#endif
+            }
+            Assert.Equal(Path.Combine(tmpDir, "BaseDir", "NLog.config"), result.First(), StringComparer.OrdinalIgnoreCase);
+            Assert.Contains(Path.Combine(tmpDir, "EntryDir", "NLog.config"), result, StringComparer.OrdinalIgnoreCase);
+            Assert.Contains(Path.Combine(tmpDir, "EntryDir", "Entry.exe.nlog"), result, StringComparer.OrdinalIgnoreCase);
+#if NETSTANDARD
+            Assert.Contains(Path.Combine(tmpDir, "EntryDir", "Entry.dll.nlog"), result, StringComparer.OrdinalIgnoreCase);
+#endif
+            Assert.Contains("NLog.dll.nlog", result.Last(), StringComparison.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void LoadConfigFile_NetCorePublished_UseProcessDirectory()
+        {
+            // Arrange
+            var d = Path.DirectorySeparatorChar;
+            var tmpDir = Path.GetTempPath();
+            var appEnvMock = new AppEnvironmentMock(f => true, f => null)
+            {
+                AppDomainBaseDirectory = Path.Combine(tmpDir, "BaseDir"),
+#if NETSTANDARD
+                AppDomainConfigurationFile = string.Empty,                  // NetCore style
+#else
+                AppDomainConfigurationFile = Path.Combine(tmpDir, "ProcessDir", "Process.exe.config"),
+#endif
+                CurrentProcessFilePath = Path.Combine(tmpDir, "ProcessDir", "Process.exe"),    // NetCore published exe
+                EntryAssemblyLocation = Path.Combine(tmpDir, "ProcessDir"),
+                EntryAssemblyFileName = "Entry.dll"
+            };
+
+            var fileLoader = new LoggingConfigurationFileLoader(appEnvMock);
+
+            // Act
+            var result = fileLoader.GetDefaultCandidateConfigFilePaths().ToList();
+
+            // Assert base-directory + process-directory + nlog-assembly-directory
+            if (NLog.Internal.PlatformDetector.IsWin32)
```

Comment:
please add some helpers for the test duplication (at least here in the assert part), thanks!

Reply:
Done